scilpy/tractanalysis/fibertube_scoring.py

```diff
@@ -437,105 +434,49 @@ def endpoint_connectivity(step_size, blur_radius, centerlines,
-            fi = indices[neighbor_segi][0]
-            if fi != seed_fi:
-                continue
+        min_dist = np.inf
+        min_seg = 0
 
-            fiber = centerlines[fi]
-            fib_end_pt1 = fiber[centerlines_length[fi] - 2]
-            fib_end_pt2 = fiber[centerlines_length[fi] - 1]
-            radius = diameters[fi] / 2
-
-            # Connectivity
-            # Is in end segment of our fibertube
-            dist, _, _, _ = dist_segment_segment(
-                fib_end_pt1, fib_end_pt2, streamline[int(np.floor(ratio))],
-                streamline[int(np.ceil(ratio))+1])
-            if dist < radius:
-                truth_connected = True
-                truth_vc.add(si)
-
-            # Resolution-wise connectivity
-            # Passes by end segment of our fibertube
-            dist, _, _, _ = dist_segment_segment(fib_end_pt1, fib_end_pt2,
-                                                 streamline[1], streamline[0])
-            if dist < radius + blur_radius:
-                res_connected = True
-                res_vc.add(si)
-
-        # If not VC we check IC/NC and if not Res_VC, we check Res_IC/Res_NC
+        # Finding closest segment
+        # There will always be a neighbor to override np.inf
         for neighbor_segi in neighbors:
             fi = indices[neighbor_segi][0]
-            if fi == seed_fi:
-                continue
-
-            fiber = centerlines[fi]
-            fib_end_pt1 = fiber[centerlines_length[fi] - 2]
-            fib_end_pt2 = fiber[centerlines_length[fi] - 1]
-            radius = diameters[fi] / 2
-
-            is_vc = len(truth_vc.intersection({si})) != 0
-            is_res_vc = len(res_vc.intersection({si})) != 0
-
-            # Connectivity
-            # Is in start or end segment of a fibertube which is not ours
-            start_dist, _, _, _ = dist_segment_segment(
-                fiber[0], fiber[1], streamline[int(np.floor(ratio))],
-                streamline[int(np.ceil(ratio))+1])
+            pi = indices[neighbor_segi][1]
 
-            end_dist, _, _, _ = dist_segment_segment(
-                fib_end_pt1, fib_end_pt2, streamline[int(np.floor(ratio))],
-                streamline[int(np.ceil(ratio))+1])
+            dist, _, _ = dist_point_segment(centerlines[fi][pi],
+                                            centerlines[fi][pi+1],
+                                            streamline[1])
 
-            if not is_vc and (start_dist < radius or end_dist < radius):
-                truth_connected = True
-                truth_ic.add(si)
+            if dist < min_dist:
+                min_dist = dist
+                min_seg = neighbor_segi
 
-            # Resolution-wise connectivity
-            # Passes by start or end segment of a fibertube which is not ours
-            start_dist, _, _, _ = dist_segment_segment(
-                fiber[0], fiber[1], streamline[1], streamline[0])
+        fi = indices[min_seg][0]
```

Comment:
Not sure I fully understand lines 468-469 and condition 472. Can you use more explicit variable names for `pi`, `fi`?